drivers/focuser/deepskydad_af3.h

```diff
@@ -0,0 +1,147 @@
+         * @return IPS_BUSY if motion is in progress. IPS_OK if motion is small and already complete. IPS_ALERT for trouble.
+         */
+        virtual IPState MoveFocuser(FocusDirection dir, int speed, uint16_t duration) override;
+
+        /**
+         * @brief MoveAbsFocuser Move to an absolute target position
+         * @param targetTicks target position
+         * @return IPS_BUSY if motion is in progress. IPS_OK if motion is small and already complete. IPS_ALERT for trouble.
+         */
+        virtual IPState MoveAbsFocuser(uint32_t targetTicks) override;
+
+        /**
+         * @brief MoveRelFocuser Move focuser for a relative amount of ticks in a specific direction
+         * @param dir Directoin of motion
+         * @param ticks steps to move
+         * @return IPS_BUSY if motion is in progress. IPS_OK if motion is small and already complete. IPS_ALERT for trouble.
+         */
+        virtual IPState MoveRelFocuser(FocusDirection dir, uint32_t ticks) override;
+
+        virtual bool SyncFocuser(uint32_t ticks) override;
+        virtual bool ReverseFocuser(bool enabled) override;
+        virtual bool AbortFocuser() override;
+        virtual void TimerHit() override;
+        virtual bool saveConfigItems(FILE * fp) override;
+
+    private:
+        bool Ack();
+        /**
+         * @brief sendCommand Send a string command to DeepSkyDad focuser.
+         * @param cmd Command to be sent in format "[CMD]"
+         * @param res If not nullptr, the function will read until it detects the response in format "(RES)"
+         *        if nullptr, no read back is done and the function returns true.
+         * @return True if successful, false otherwise.
+         */
+        bool sendCommand(const char * cmd, char * res = nullptr);
+        bool sendCommandSet(const char * cmd);
+
+        // Get initial focuser parameter when we first connect
+        void GetFocusParams();
+        bool readStepMode();
+        bool readSpeedMode();
+        bool readPosition();
+        bool readMaxPosition();
+        bool readMaxMovement();
+        bool readSettleBuffer();
+        bool readMoveCurrentMultiplier();
+        bool readHoldCurrentMultiplier();
+        bool isMoving();
+        bool readTemperature();
+
+        void timedMoveCallback();
+
+        bool MoveFocuser(uint32_t position);
+
+        double targetPos { 0 }, lastPos { 0 }, lastTemperature { 0 };
+
+        // Step mode
+        ISwitch StepModeS[9];
+        ISwitchVectorProperty StepModeSP;
+
+        // Speed mode
+        ISwitch SpeedModeS[5];
+        ISwitchVectorProperty SpeedModeSP;
+
+        //Current move 
+        INumber MoveCurrentMultiplierN[1];
+        INumberVectorProperty MoveCurrentMultiplierNP;
+
+        //Current hold
+        INumber HoldCurrentMultiplierN[1];
+        INumberVectorProperty HoldCurrentMultiplierNP;
+
+        // Max movement
+        INumber FocusMaxMoveN[1];
+        INumberVectorProperty FocusMaxMoveNP;

```

Comment:
There is no need to define this property since it is defined by the INDI::FocuserInterface already (FocusMaxPosNP). So instead just override the function bool SetFocuserMaxPosition(uint32_t ticks) and implement the functionality there.

Reply:
Hi Jasem, thank you!

I am not sure I understand.

_"There is no need to define this property since it is defined by the INDI::FocuserInterface already (FocusMaxPosNP)"_

I do not have FocusMaxPosNP defined in the header. I have only defined FocusMaxMoveN, as it is not defined in FocuserInterface?

Reply:
Just to clarify that I understand this correctly:

Maximum position (FocusMaxPosNP) is the biggest value that motor can be at. e.g. 1000000?

Maximum move is maximum change of position in single movement and I did not find this in the focuserinterface. So I added FocusMaxMoveNP.

Is there already a property for maximum number of steps in a single move?

Reply:
Ah okay, I got the two confused then. Yes, you understanding is correct.